src/datatypes/validation/gadget/PoisonPills.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file   PoisonPills.cpp
+ * @author Marco Deuscher
+ * @date   28.04.2020 (creation)
+ * @brief  Implemenation of poison pills gadget validation.
+ */
+
+#include "datatypes/gameplay/Movement.hpp"
+#include "GadgetValidator.hpp"
+
+namespace spy::gameplay {
+    bool GadgetValidator::validatePoisonPills(const State &s, GadgetAction a) {
+        using spy::gadget::GadgetEnum;
+
+        // check if target inside map
+        if (!s.getMap().isInside(a.getTarget())) {
+            return false;
+        }
+
+        // check if character has poisonous pills
+        auto character = s.getCharacters().findByUUID(a.getCharacterId().value());
+        auto gadgets = character->getGadgets();
+        auto gadget = std::find_if(gadgets.begin(), gadgets.end(), [](const gadget::Gadget &g) {
+            return g.getType() == GadgetEnum::POISON_PILLS;
+        });
+
+        // check if target contains cocktail
+        bool targetHasCocktail = false;
+        auto targetFieldGadget = s.getMap().getField(a.getTarget()).getGadget();
+        if (targetFieldGadget.has_value()) {
+            targetHasCocktail = targetFieldGadget->getType() == GadgetEnum::COCKTAIL;
```

Comment:
Das würde ich der Übersichtlichkeit halber auch nochmal klammern
```suggestion
            targetHasCocktail = (targetFieldGadget->getType() == GadgetEnum::COCKTAIL);
```